transmart-api-server/keycloak.md

```diff
@@ -0,0 +1,72 @@
+# How to set up transmart to work with keycloak.
+
+##Keycloak configuration
+
+Below are steps on how to set keycloak for transmart using the admin console.
+The prerequisite is to have admin credentials to a keycloak instance. https://<domain>/auth/admin/
+
+1. Create a realm. e.g. Dev
+2. Create a client. e.g. transmart
+
+    - Client Protocol: openid-conect
+    - Access Type: confidential
+    - Standard Flow Enabled: On
+    - Valid Redirect URIs: https://<client url. e.g. glowing bear>
+    - Web Origins: https://<client url. e.g. glowing bear>
+    
+3. Create Roles.  
+    **Note:** not realm roles, but client roles.
+    Follow Clints > Roles (tab)
+    
+    -Roles
+    
+        ROLE_ADMIN
+        
+    -Study permissions
+    
+        SHOW_SUMMARY_STATISTICS|EXP:SCSCP
+        READ|EXP:SCSCP
+        
+    The convention is to represent study permissions as `operation|study token`
```

Comment:
Let's discuss if `permisson|study token` is not a better approach.

Reply:
Yes, `Permission` seems to be a good candidate for this.